makefile

```diff
@@ -0,0 +1,67 @@
+#
+# The root makefile
+#
+
+OUTPUT_PATH=out
+OUTPUT_NAME=test
+
+# The cross compile if exists, otherwise uses the gcc
+CROSS_COMPILE?=
+
+#
+# Source (.c) Files
+#
+SOURCE_FILES= \
+	Source/test.c
+
+#
+# Header (.h) File Directories
+#
+#  Bear in mind, include directories must have -I prefix
+#
+INCLUDE_DIRECTORIES= \
+	-IInclude/
+
+#
+# Project Definitions
+#
+#  Bear in mind, definitions must have -D prefix
+#
+DEFINES= \
+	-DTEST_CONFIG=255
+
+#
+# All Flags
+#
+CFLAGS=${INCLUDE_DIRECTORIES} ${DEFINES}
+
+
+#############################################
+# Rules
+#############################################
+
+
+#
+# The main rule
+#
+all: out test
+
+# Rebuild
+rebuild: clean all
+
+# The test rule
+test:
```

Comment:
'make test' will fail if run before 'make out'. I think all rules should work standalone.
In this case we may bind 'out' rule to 'test' rule. @zaya-mc

Reply:
Yes it is just test rule for make all but ok I will fix.